traffic_ops/traffic_ops_golang/invalidationjobs/invalidationjobs.go

```diff
@@ -0,0 +1,928 @@
+		"deliveryService": dbhelpers.WhereColumnInfo{`(SELECT deliveryservice.xml_id FROM deliveryservice WHERE deliveryservice.id=job.job_deliveryservice)`, nil},
+		"dsId":            dbhelpers.WhereColumnInfo{"job.job_deliveryservice", api.IsInt},
+	}
+
+	where, orderBy, queryValues, errs := dbhelpers.BuildWhereAndOrderBy(job.APIInfo().Params, queryParamsToSQLCols)
+	if len(errs) > 0 {
+		var b strings.Builder
+		b.WriteString("Reading jobs:")
+		for _, err := range errs {
+			b.WriteString("\n\t")
+			b.WriteString(err.Error())
+		}
+
+		return nil, nil, errors.New(b.String()), http.StatusInternalServerError
+	}
+
+	// TODO: check tenancy here
+	query := readQuery + where + orderBy
+
+	log.Debugln("generated job query: " + query)
+	log.Debugf("executing with values: %++v\n", queryValues)
+
+	returnable := []tc.InvalidationJob{}
+	rows, err := job.APIInfo().Tx.NamedQuery(query, queryValues)
+	if err != nil {
+		return nil, nil, fmt.Errorf("querying: %v", err), http.StatusInternalServerError
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		j := tc.InvalidationJob{}
+		err := rows.Scan(&j.ID,
+			&j.Keyword,
+			&j.Parameters,
+			&j.AssetURL,
+			&j.StartTime,
+			&j.CreatedBy,
+			&j.DeliveryService)
+		if err != nil {
+			return nil, nil, fmt.Errorf("parsing db response: %v", err), http.StatusInternalServerError
+		}
+
+		returnable = append(returnable, j)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, nil, fmt.Errorf("Parsing db responses: %v", err), http.StatusInternalServerError
+	}
+
+	// This cannot be done in the scanning loop, because pq will throw an error if you try to make
+	// another query before exhausting the rows returned by an earlier query
+	filtered := []interface{}{}
+	for _, r := range returnable {
+		userErr, sysErr, errCode := IsUserAuthorizedToModifyDSXMLID(job.APIInfo(), *r.DeliveryService)
+		if sysErr != nil {
+			return nil, userErr, sysErr, errCode
+		} else if userErr == nil {
+			filtered = append(filtered, r)
+		}
+	}
+
+	return filtered, nil, nil, http.StatusOK
+}
+
+// Used by POST requests to `/jobs`, creates a new content invalidation job
+// from the provided request body.
+func Create(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	job := tc.InvalidationJobInput{}
+	cType := r.Header.Get(http.CanonicalHeaderKey("content-type"))
+	if cType == tc.ApplicationJson || cType == "" {
+		decoder := json.NewDecoder(r.Body)
+		if err := decoder.Decode(&job); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("Unable to parse Invalidation Job"), fmt.Errorf("parsing jobs/ POST: %v", err))
+			return
+		}
+	} else {
+		w.Header().Set(http.CanonicalHeaderKey("accept"), strings.Join([]string{tc.ApplicationJson, "application/x-www-form-urlencoded", "multipart/form-data"}, ";"))
```

Comment:
I think this is returning to the user this is the headers we support on content? (ignore if not) It says here we support `"application/x-www-form-urlencoded", "multipart/form-data"` but this handler will error out if it is not ApplicationJson?

Reply:
Oh, that was a part of an earlier, misguided effort to make this the place where content negotiation takes place. It shouldn't still be in there.